dovecot.service.in

```diff
@@ -30,8 +30,10 @@ ExecReload=@bindir@/doveadm reload
 ExecStop=@bindir@/doveadm stop
 PrivateTmp=true
 NonBlocking=yes
-# Enable this if your systemd is new enough to support it:
-#ProtectSystem=full
+ProtectSystem=full
+NoNewPrivileges=true
+PrivateDevices=true
+CapabilityBoundingSet=CAP_CHOWN CAP_DAC_OVERRIDE CAP_IPC_LOCK CAP_KILL CAP_NET_BIND_SERVICE CAP_SETGID CAP_SETUID CAP_SYS_CHROOT
```

Comment:
After testing it, it was missing CAP_SYS_RESOURCE. Seems that NoNewPrivileges at least initially works with environment with multiple UIDs for users.
```
Sep 19 09:03:52 cypher dovecot[10585]: master: Error: setrlimit(RLIMIT_NPROC, 4390): Operation not permitted
```